src/container/srv_layout.c

```diff
@@ -83,17 +86,104 @@ struct daos_prop_entry cont_prop_entries_default[CONT_PROP_NUM] = {
 		.dpe_val	= 0, /* No limitation */
 	}, {
 		.dpe_type	= DAOS_PROP_CO_ACL,
-		.dpe_val_ptr	= NULL,
+		.dpe_val_ptr	= NULL, /* generated dynamically */
 	}, {
 		.dpe_type	= DAOS_PROP_CO_COMPRESS,
 		.dpe_val	= DAOS_PROP_CO_COMPRESS_OFF,
 	}, {
 		.dpe_type	= DAOS_PROP_CO_ENCRYPT,
 		.dpe_val	= DAOS_PROP_CO_ENCRYPT_OFF,
+	}, {
+		.dpe_type	= DAOS_PROP_CO_OWNER,
+		.dpe_str	= "NOBODY@",
+	}, {
+		.dpe_type	= DAOS_PROP_CO_OWNER_GROUP,
+		.dpe_str	= "NOBODY@",
 	}
 };
 
 daos_prop_t cont_prop_default = {
 	.dpp_nr		= CONT_PROP_NUM,
 	.dpp_entries	= cont_prop_entries_default,
 };
+
+/**
+ * The default ACL includes ACEs for owner and the assigned group. All others
+ * are denied by default.
+ */
+static struct daos_ace *
+alloc_ace_with_access(enum daos_acl_principal_type type, uint64_t permissions)
```

Comment:
Is this container specific? It looks like you could put this into the security folder and use it both in here and in the pool code as well.

Reply:
We absolutely can. Good catch.

Reply:
We could move the default ACL generation functions to the security module as well. It is different for pools vs. containers but might be good to centralize that code in the security module.